### PR TITLE
Impro-672 use realizations rather than members

### PR DESCRIPTION
**Pull request by markysparks**
Addresses #543

Replace instances within Improver of `members` and `member` with the CF compliant ensemble terms `realizations` and `realization`, including updating unit tests. 

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py [hunk 1]
```diff
@@ -705,83 +705,85 @@ def __init__(self):
...
-        members within the raw ensemble forecast is random, such that the
-        raw ensemble members are exchangeable. If fewer percentiles are
-        requested than ensemble members, then only the first n ensemble
-        members are used.
+        of percentiles and the number of raw forecast realizations. If more
+        percentiles are requested than ensemble realizations, then the ensemble
+        realizations are recycled. This assumes that the identity of the
+        ensemble realizations within the raw ensemble forecast is random, such
+        that the raw ensemble realizations are exchangeable. If fewer
+        percentiles  are requested than ensemble realizations, then only the
```
Comment (PaulAbernethy) [inline lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py]: There is a double-space on this line @ 'percentiles  are requested'

File: tests/improver-nbhood-land-and-sea/01-help.bats [hunk 2]
```diff
@@ -97,6 +98,7 @@ Neighbourhooding Radius - Set only one of the options:
                         For example: 10000,12000,14000 1,2,3 where a lead time
                         of 1 hour uses a radius of 10000m, a lead time of 2
                         hours uses a radius of 12000m, etc.
+
```
Comment (PaulAbernethy) [inline tests/improver-nbhood-land-and-sea/01-help.bats]: Extra blank line has been introduced here.

File: tests/improver-nbhood-vicinity/01-help.bats [hunk 3]
```diff
@@ -70,15 +70,17 @@ optional arguments:
...
+                        = 1.0 this essentially conserves ensemble realizations
+                        if every grid square is considered to be the
+                        equivalent of an ensemble realization.Optional,
+                        defaults to 1.0.
   --weighted_mode       For neighbourhood processing using a circular kernel,
                         setting the weighted_mode indicates the weighting
                         decreases with radius. If weighted_mode is not set, a
                         constant weighting is assumed. Currently this keyword
                         does nothing as only a square kernel is applicable.
+
```
Comment (PaulAbernethy) [inline tests/improver-nbhood-vicinity/01-help.bats]: Extra blank line has been introduced here.

File: tests/improver-percentile/01-help.bats [hunk 4]
```diff
@@ -68,6 +68,7 @@ optional arguments:
                         Optional definition of the number of percentiles to be
                         generated, these distributed regularly with the aim of
                         dividing into blocks of equal probability.
+
```
Comment (PaulAbernethy) [inline tests/improver-percentile/01-help.bats]: Extra blank line has been introduced here.

File: bin/improver-ecc [hunk 5]
```diff
@@ -110,11 +111,12 @@ def main():
 
     rebadging = parser.add_argument_group(
         'Rebadging options', 'Options for rebadging the input percentiles '
-        'as ensemble members.')
-    rebadging.add_argument('--member_numbers', default=None,
-                           metavar='MEMBER_NUMBERS',
-                           help='A list of ensemble member numbers to use '
-                           'when rebadging the percentiles into members.')
+        'as ensemble s.')
```
Comment (gavinevans) [inline bin/improver-ecc]: I think that the `s` should be `realizations`.

File: tests/improver-nbhood-land-and-sea/01-help.bats [hunk 6]
```diff
@@ -61,10 +61,11 @@ optional arguments:
...
   --ens_factor ENS_FACTOR
                         The factor with which to adjust the neighbourhood size
-                        for more than one ensemble member. If ens_factor = 1.0
-                        this essentially conserves ensemble members if every
-                        grid square is considered to be the equivalent of an
-                        ensemble member.Optional, defaults to 1.0.
+                        for more than one ensemble realization. If ens_factor
+                        = 1.0 this essentially conserves ensemble realizations
+                        if every grid square is considered to be the
+                        equivalent of an ensemble realization.Optional,
```
Comment (gavinevans) [inline tests/improver-nbhood-land-and-sea/01-help.bats]: There should probably be a space after `ensemble realization` i.e. `ensemble realization. Optional`

File: tests/improver-nbhood-vicinity/01-help.bats [hunk 7]
```diff
@@ -70,15 +70,17 @@ optional arguments:
...
   --ens_factor ENS_FACTOR
                         The factor with which to adjust the neighbourhood size
-                        for more than one ensemble member. If ens_factor = 1.0
-                        this essentially conserves ensemble members if every
-                        grid square is considered to be the equivalent of an
-                        ensemble member.Optional, defaults to 1.0.
+                        for more than one ensemble realization. If ens_factor
+                        = 1.0 this essentially conserves ensemble realizations
+                        if every grid square is considered to be the
+                        equivalent of an ensemble realization.Optional,
```
Comment (gavinevans) [inline tests/improver-nbhood-vicinity/01-help.bats]: There should probably be a space after `ensemble realization` i.e. `ensemble realization. Optional`

**Pull request merged**